ion-schema/src/violation.rs

```diff
@@ -86,12 +86,30 @@ impl Violation {
     }
 }
 
-// TODO: Implement Violation with proper indentation for the nested tree of violations
 impl fmt::Display for Violation {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "A validation error occurred: {}", self.message)?;
-        for v in &self.violations {
-            write!(f, "  {v}")?;
+        f.write_str(self.message.as_str())?;
+
+        let mut stack = vec![];
+        let mut violations_iter = self.violations.iter();
+        let mut violation = violations_iter.next();
+
+        while let Some(v) = violation {
+            f.write_fmt(format_args!(
+                "\n{}- {}",
+                "  ".repeat(stack.len() + 1),
```

Comment:
Consider adding a `TODO` for a reusable indentation `String` rather than re-allocating/formatting one in each iteration.